src/QtAVPlayer/qavplayer.cpp

```diff
@@ -462,6 +463,12 @@ void QAVPlayer::seek(qint64 pos)
     if (pos < 0 || (duration() > 0 && pos > duration()))
         return;
 
+    // allow to seek only if media loaded / valid
+    if(d->mediaStatus != QAVPlayer::LoadedMedia && d->mediaStatus != QAVPlayer::EndOfMedia && d->mediaStatus != QAVPlayer::SeekingMedia)
+        return;
+
+    d->setMediaStatus(QAVPlayer::SeekingMedia);
```

Comment:
also it would be good to have a test which covers seeking media.

Reply:
okay, will try to add it

Reply:
Hmm, seems like cmake project doesn't include tests. From the other side existing qmake project disallows custom ffmpeg usage and pretty inconvenient to use (at least on windows)